plugins/flytekit-duckdb/flytekitplugins/duckdb/task.py

```diff
@@ -44,6 +99,25 @@ def __init__(
             **kwargs,
         )
 
+    def _connect_to_duckdb(self):
+        """
+        Handles the connection to DuckDB based on the provider.
+
+        Returns:
+            A DuckDB connection object.
+        """
+        connect_token = None
+        if self._connect_secret:
+            connect_token = current_context().secrets.get(
+                group=self._connect_secret.group,
+                key=self._connect_secret.key,
+                group_version=self._connect_secret.group_version,
+            )
+        if isinstance(self._provider, DuckDBProvider):
+            return self._provider.value(connect_token)
+        else:  # callable
+            return self._provider(connect_token)
```

Comment:
how about we not send the token when there isn't one provided?

Reply:
can you also remove the token param from the connect_local function?

Reply:
sure, I removed the token and gave the option for callables that don't have tokens